lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.h

```diff
@@ -981,4 +981,11 @@ class SwiftASTContextForExpressions : public SwiftASTContext {
 };
 
 } // namespace lldb_private
+
+#ifndef NDEBUG
+/// Provided only for unit tests.
+template <typename SmallString>
+void ApplyWorkingDir(SmallString &clang_argument, llvm::StringRef cur_working_dir);
```

Comment:
c++ question: should this have an explicit instantiation? or does it not matter?

Reply:
this was discussed offline, the conclusion has been: explicit instantiation not needed

Reply:
This works. I originally had in mind a static member function of SwiftASTContext, but this is equivalent.

Instead of making this a template, I think we can use the base class `SmallStringImpl &` here.

Reply:
Actually, this doesn't work. Now release builds won't be able to build the unit test target. Can we make it a static member of Swift ASTContext instead?

Reply:
This was based on the existing code further up the header:

https://github.com/apple/llvm-project/blob/swift/main/lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.h#L176-L179

Would you rather the `NDEBUG` guard is not present? I'll update the function to be a static member fn and remove the template.

Reply:
Yes, but the test that depends on this constructor is also guarded by a `#ifndef NDEBUG` (line 195).
If we make the function a static member of SwiftASTContext, we can leave the tests unguarded. I think that's the cleanest approach.

Reply:
I don't see a `SmallStringImpl` base class anywhere though, it appears to inherit directly from `SmallVector`. Fine with keeping this templated for now?

Reply:
You're right, it should have been SmallVectorImpl, not SmallStringImpl. Here's an example API that works like this:
https://www.llvm.org/doxygen/namespacellvm_1_1sys_1_1fs.html#a2363fbfbd0a348f5d81fc3d3223ecae3

If using the underlying SmallVectorImpl is too cumbersome, I would just hardcode the buffer size instead of using a template. But it looks like we are only converting clang_argument to a StringRef and append characters to it.

Reply:
I see, yes, that works out fine, I've updated it now.